src/components/input/SingleLineInputBase.js

```diff
@@ -117,6 +118,8 @@ SingleLineInputBase.propTypes = {
   warning: PropTypes.oneOfType([PropTypes.bool, PropTypes.node]),
   /** A custom width for the input field */
   width: PropTypes.string,
+  /** Whether to disable styling that hints being able to type in the input field */
+  noInputStyling: PropTypes.bool,
```

Comment:
PP: rename this to what it does; f.e. a variable called `typeable` which is set to `true` by default

Reply:
The thing is, it's a bit of a weird case; 
it doesn't actually prevent typing in it, it just hides the styling that makes it look like  you can type in it
This all stems from the issue that we're using an Input field here, but don't use the input part of it
So I wanted to put an emphasis on the _styling_ part so that it doesn't seem like it would disable the typing behaviour

Reply:
So there's a case where we want to not show the styling which suggests the input is typeable, but allow users to type in it? Can we not use the input and create a different input for it in that case? Just trying to avoid adding props like these because the issue seems to be somewhere else and our case just doesn't fit this abstraction

Reply:
Yeah, we were thinking "button" might be a better fit here, but otherwise we considered this another step below disabled and read-only, where you can't input anything, but you can still click in it (and thus trigger the datepicker)
I think the whole issue really just stems from the use of Input, and I don't know why that choice was made, so I can't just change it. Design just asked to remove the stuff that makes it look like it's editable